### PR TITLE
fdtrack: hack to track file descriptor usage

### PR DESCRIPTION
**Pull request by fjl**
The last commit in this PR is meant to be temporary (to be reverted in the next release).
It adds tracking of open files and sockets in a few parts of the codebase,
including third-party code in Godeps. Package fdtrack produces a line like the one below every 15s.

```
I0803 16:02:23.163301   51094 fdtrack.go:109] fd usage 91/256, tracked 58 [ipc:1 leveldb:31 p2p:26]
```

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Mon Aug 3 01:24:44 UTC 2015_

File: rpc/comms/ipc_unix.go [hunk 1]
```diff
@@ -50,15 +51,16 @@ func (self *ipcClient) reconnect() error {
 func startIpc(cfg IpcConfig, codec codec.Codec, api shared.EthereumApi) error {
 	os.Remove(cfg.Endpoint) // in case it still exists from a previous run
 
-	l, err := net.ListenUnix("unix", &net.UnixAddr{Name: cfg.Endpoint, Net: "unix"})
+	l, err := net.Listen("unix", cfg.Endpoint)
```
Comment (obscuren) [inline rpc/comms/ipc_unix.go]: Is this change required for `fdtrack`?

Comment (fjl) [inline rpc/comms/ipc_unix.go]: Yes. Using `ListenUnix` is sort-of silly anyway if the additional methods aren't used.

**Comment by fjl** [comment]
I found an issue with this PR. Please don't merge yet.

**Comment by obscuren** [comment]
Please also squash that which will stay and move the temporary code in another commit.

**Comment by fjl** [comment]
I did. The last commit is the one to be reverted.

**Comment by fjl** [comment]
I'll resurrect some of the code from package `fdtrack` for #1318 after 1.0.1.

**Comment by fjl** [comment]
I have fixed the issue. The issue was that setting large limit could cause a huge allocation on OS X.

File: rpc/comms/http.go [hunk 2]
```diff
@@ -31,10 +37,15 @@ import (
 	"github.com/rs/cors"
 )
 
+const (
+	serverIdleTimeout  = 10 * time.Second // idle keep-alive connections
+	serverReadTimeout  = 15 * time.Second // per-request read timeout
+	serverWriteTimeout = 15 * time.Second // per-request read timeout
```
Comment (bas-vk) [inline rpc/comms/http.go]: copy paste error in comment

**Comment by obscuren** [comment]
:+1: fd part.

I'll defer to @bas-vk judgement for the RPC interface changes.

**Comment by bas-vk** [comment]
:+1:
